src/routeGeneration/templates/express.hbs

```diff
@@ -24,16 +24,20 @@ const models: TsoaRoute.Models = {
     {{#each models}}
     "{{@key}}": {
         {{#if enums}}
+        "dataType": "refEnum",
         "enums": {{{json enums}}},
         {{/if}}
         {{#if properties}}
+        "dataType": "refObject",
         "properties": {
             {{#each properties}}
             "{{@key}}": {{{json this}}},
             {{/each}}
         },
         {{/if}}
-        "additionalProperties": {{additionalPropsHelper additionalProperties}},
+        {{#if additionalProperties}}
```

Comment:
validateModel grabs this [here](https://github.com/lukeautry/tsoa/blob/master/src/routeGeneration/templateHelpers.ts#L632).

Afterwards it checks if it `=== false`.
We should make it either check falsy of put false in the template by default. This causes the bug in validateModel and therefore breaks validate Union which calls validateModel.

Reply:
I am checking falsy. It's just unclear because I was being fancy/tricky with it because I included that check inside of the `isDefaultForAdditionalPropertiesAllowed` function. What I was trying to do was to use "code as comments" to clarify that an undefined value for `additionalProperties` should be construed as "yes, additional properties are allowed."

So if I make your suggested change, then that  would kind of subvert the goal of that method.

I opted instead to make sure that the `"additionalProperties"` property is always there in the template, but that it's value might be literally `undefined`. That makes this fix a handlebars only fix.

Thank you so much for pointing me in the right direction! :)